src/App.tsx

```diff
@@ -1,24 +1,183 @@
+import { Footer } from './components/Footer';
+import { ErrorTypes } from './types/ErrorTypes';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TodoItem } from './components/TodoItem';
 
-const USER_ID = 0;
+const USER_ID = 10905;
 
 export const App: React.FC = () => {
+  const [title, setTitle] = useState<string>('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedFilter, setSelectedFilter] = useState('All');
+  const [loadingTodosId, setLoadingTodosId] = useState<number[]>([0]);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [errorMessage, setErrorMessage] = useState<ErrorTypes>();
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  const handleChangeTitle = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(e.target.value);
+  };
+
+  const showAndDeleteError = () => {
+    setIsError(true);
+    setTimeout(() => setIsError(false), 3000);
+  };
+
+  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (!title) {
+      setErrorMessage(ErrorTypes.TITLE);
+      showAndDeleteError();
+    }
+
+    setIsDisabled(true);
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title,
+      completed: false,
+    });
+
+    addTodos({
+      title,
+      userId: USER_ID,
+      completed: false,
+    })
+      .then((res) => {
+        setTodos(prevTodos => [...prevTodos, res]);
+        setIsDisabled(false);
+        setTempTodo(null);
+      })
+      .catch(() => {
+        setErrorMessage(ErrorTypes.ADD);
+        showAndDeleteError();
+        setIsDisabled(false);
+        setTempTodo(null);
+      });
+
+    setTitle('');
+  };
+
+  const filterTodos = (filterTodosBy: string) => {
+    switch (filterTodosBy) {
+      case 'Active':
+        return todos.filter(todo => !todo.completed);
+
+      case 'Completed':
+        return todos.filter(todo => todo.completed);
+
+      case 'All':
+      default:
+        return todos;
+    }
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    setLoadingTodosId(prevTodos => [...prevTodos, todoId]);
+    deleteTodos(todoId)
+      .then(() => {
+        setTodos(prevTodos => [...prevTodos
+          .filter(todo => todo.id !== todoId)]);
+        setLoadingTodosId(prevLoadingTodos => [...prevLoadingTodos
+          .filter(Id => Id !== todoId)]);
+      })
+      .catch(() => {
+        setErrorMessage(ErrorTypes.DELETE);
+        showAndDeleteError();
+        setLoadingTodosId(prevLoadingTodos => [...prevLoadingTodos
+          .filter(Id => Id !== todoId)]);
+      });
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setErrorMessage(ErrorTypes.LOAD);
+        showAndDeleteError();
+      });
+  }, []);
+
+  useEffect(() => {
+    if (isDisabled) {
+      setTimeout(() => {
```

Comment:
It's better to avoid nesting for better readability
```
if (!isDisabled) {
  return;
} 
....

Reply:
done